Indexing/IndexManager.hpp

```diff
@@ -98,6 +99,8 @@ class IndexManager
   DHMap<IndexType,Entry> _store;
 
   Index* create(IndexType t);
+  Shell::Options::UnificationWithAbstraction _uwa;
+  bool _uwaFixedPointIteration;
```

Comment:
We cannot quite use that. For `_uwaFixedPointIteration` we can, but not for `_uwa` as we have this interplay between the options `-fe abstraction` vs `-uwa function_extensionality` and then additionally the constraint that function extensionality abstaction only being enabled when the problem is higher order. Soooo I'm not quite sure what to do. I think the big question is, do we care about the current higher order schedules, or do we drop them anyways because there's ahmed's better higher order branch, and we don't maintain our higher order stuff until that get's merge and we replace the master stuff in favor of ahmed's new things. @ibnyusuf @quickbeam123 , what do you think?

Reply:
Higher-order support on `ahmed-new-hol` is an immense improvement on what we have in current master. However, it also conflicts with a lot of what is in this PR, so it will be a large task to get it in on top of this.

My thinking is, @joe-hauns , if you feel confidant that you will be able to merge in the new higher-order branch down th eline then it is fine to break current higher-order support and modify the compiler to reject all higher-order problems. This would be temporary until `ahmed-new-hol` is merged.

On the other hand, if you are not sure whether you will get around to merging `ahmed-new-hol`, then breaking current HOL support on `master ` would leave Vampire without higher-order support on master indefinitely.

I will be able to assist with merging in `ahmed-new-hol` if you go down that route.

Reply:
So I think that branch we already talked about at some point this summer, right? I think the plan was then to reimplement it as it diverged too much from master, right? So my plan was & is to take on this work (and of course work with you on that @ibnyusuf ). To that i committed to in the dev meeting the other week in vienna. There we said we'd keep the current branch `ahmed-new-hol` (assuming this is what entered last years CASC) as the go-to higher order branch until i did the merging/reimplementing.

So if this is the plan anyways we do not need to bother about the old HOL things that are in this PR too much, as we are going to replace them by other stuff and have a better HOL branch as well.  (Of course I'm not suggesting to just kick everything out but I mean not bothering about little changes in which options are set and so on).

Reply:
Great, in that case I'd say break current higher-order support in master to your heart's content! 

The safest thing to do would be to add a few lines of code to the parser to reject higher-order problems perhaps with a message to say that higher-order support will be returning in the near future. I don't know that we have any users of higher-order Vampire, but in case anyone does try it, that will be a lot easier to understand than failing in a subtle, difficult to understand manner.

Reply:
@hetzenmat may be interested on working on Vampire's HOL support part-time: I'll try to organise a meeting in the last week of January with Andrei so we can discuss how to work on this.